src/mlpack/methods/ann/gan.hpp

```diff
@@ -99,15 +95,33 @@ class GAN
   void Train(OptimizerType& Optimizer);
 
   /**
-   * Evaluate function for the GAN gives the performance of the GAN on the
-   * current input.
+   * Evaluate function for the GAN.
+   * This function gives the performance of the GAN on the current input.
    *
    * @param parameters The parameters of the network.
    * @param i Index of the current input.
+   * @param batchSize Variable to store the present number of inputs.
    */
-  double Evaluate(const arma::mat& parameters,
-                  const size_t i,
-                  const size_t batchSize);
+  template<typename Policy = PolicyType>
+  typename std::enable_if<std::is_same<Policy, StandardGAN>::value ||
+                          std::is_same<Policy, DCGAN>::value, double>::type
+  Evaluate(const arma::mat& parameters,
+           const size_t i,
+           const size_t batchSize);
+
+  template<typename Policy = PolicyType>
+  typename std::enable_if<std::is_same<Policy, WGAN>::value,
+                          double>::type
+  Evaluate(const arma::mat& parameters,
```

Comment:
Can you add a comment for the other Evaluate forms as well, I know this is redundant, but doxygen will pick this one up but unfortunately doxygen isn't going to use the first comment.